.internal/copyArrayWithOwnProperty.js

```
@@ -0,0 +1,21 @@
+/**
+ * Copies the values of `source` to `array`.
+ *
+ * @private
+ * @param {Array} source The array to copy values from.
+ * @param {Array} [array=[]] The array to copy values to.
+ * @returns {Array} Returns `array`.
+ */
+function copyArrayWithOwnProperty(source, array) {
+  const length = source.length
```

const length = source?.length ?? 0

---

I fixed it, but I think it’s better to judge by Array.isArray

---

What do you think?

---

Yes, it seems to be more appropriate for the said scenario. And in case the source is not array we should throw an exception. Thoughts ?

---

`!Array.isArray(source) && throw new Error('source is not an array!') ;`

---

Now , in the suggested workflow , it will throw an exception if the supplied argument is not an array.